TrueDialog.Client.Standard/Client/Context/Account/AccountMediaContext.cs

```diff
@@ -0,0 +1,166 @@
+
+            return new MediaContent
+            {
+                Content = content,
+                ContentType = contentType
+            };
+        }
+
+        public List<AccountMedia> GetList(int accountId, bool throwIfEmpty = false)
+        {
+            return Api.Get<List<AccountMedia>>($"/account/{accountId}/image", throwIfEmpty);
+        }
+
+        public List<AccountMedia> GetList(bool throwIfEmpty = false)
+        {
+            return GetList(CurrentAccount, throwIfEmpty);
+        }
+
+        public AccountMedia GetById(int accountId, int imageId, bool throwIfEmpty = false)
+        {
+            return Api.Get<AccountMedia>($"/account/{accountId}/image/{imageId}", throwIfEmpty);
+        }
+
+        public AccountMedia GetById(int imageId, bool throwIfEmpty = false)
+        {
+            return GetById(CurrentAccount, imageId, throwIfEmpty);
+        }
+
+        public AccountMedia UploadAsByteArray(MediaType type, byte[] byteArray)
+        {
+            return UploadAsByteArray(CurrentAccount, type, byteArray);
+        }
+
+        public AccountMedia UploadAsByteArray(string extension, byte[] byteArray)
+        {
+            return UploadAsByteArray(CurrentAccount, TypeFromExtension(extension), byteArray);
+        }
+
+        public AccountMedia UploadAsByteArray(int accountId, MediaType type, byte[] byteArray)
+        {
+            var extension = ExtensionFromMimeType(type);
+            var fileName = String.Format("ACCT{0}_{1}_Media{2}", accountId, DateTime.UtcNow.Ticks, extension);
+            var contentType = ContentTypeFromMime(type);
+
+            return Api.Upload<AccountMedia>($"/account/{accountId}/image", byteArray, contentType, fileName);
+        }
+
+        public AccountMedia UploadAsBase64(int accountId, string base64String)
+        {
+            var extension = ExtensionFromMimeType(base64String);
+            var type = TypeFromExtension(extension);
+            var bytes = BytesFromBase64String(base64String);
+
+            return UploadAsByteArray(accountId, type, bytes);
+        }
+
+        private byte[] BytesFromBase64String(string base64string)
+        {
+            CompareInfo compareInfo = CultureInfo.InvariantCulture.CompareInfo;
+            int index = compareInfo.IndexOf(base64string, ";base64,", CompareOptions.Ordinal);
+            if (index < 0)
+            {
+                return null;
+            }
+            var base64clean = base64string.Substring(index + ";base64,".Length);
+
+            return Convert.FromBase64String(base64clean);
+        }
+
+        private string ContentTypeFromMime(MediaType type)
+        {
+            switch (type)
+            {
+                case MediaType.ImageGif:
+                    return "image/gif";
+                case MediaType.ImageJpg:
+                    return "image/jpg";
+                case MediaType.ImagePng:
+                    return "image/png";
+            }
+
+            return "";
```

Comment:
Just FYI, if this is .NET Standard 3.1 with C# 7 (8?) or later you can use a value based switch:
```csharp
return type switch
{
    MediaType.ImageGif => "image/gif",
    MediaType.ImageJpg => "image/jpg",
    MediaType.ImagePng => "image/png",
    _ => String.Empty
};
```

Reply:
Not available in C# 7.3